api/client/client/client_test.go

```diff
@@ -6,217 +6,39 @@ package client_test
 import (
 	"bytes"
 	"encoding/json"
-	"fmt"
 	"io"
 	"net/http"
-	"net/url"
 	"strings"
 	"time"
 
 	"github.com/golang/mock/gomock"
 	"github.com/gorilla/websocket"
 	"github.com/juju/errors"
-	"github.com/juju/loggo"
-	"github.com/juju/names/v4"
 	"github.com/juju/testing"
 	jc "github.com/juju/testing/checkers"
 	gc "gopkg.in/check.v1"
 
 	"github.com/juju/juju/api"
 	basemocks "github.com/juju/juju/api/base/mocks"
 	"github.com/juju/juju/api/client/client"
-	"github.com/juju/juju/api/client/client/mocks"
-	"github.com/juju/juju/api/common"
 	apiservererrors "github.com/juju/juju/apiserver/errors"
-	"github.com/juju/juju/rpc"
 	"github.com/juju/juju/rpc/params"
 )
 
-type clientSuite struct {
-	Connection *mocks.MockConnection
-	Ctrl       *gomock.Controller
-}
+type clientSuite struct{}
 
 var _ = gc.Suite(&clientSuite{})
 
 // TODO(jam) 2013-08-27 http://pad.lv/1217282
 // Right now most of the direct tests for client.Client behavior are in
```

Comment:
has this todo been resolved?

Reply:
I'm not sure about this, because the [referenced bug](https://bugs.launchpad.net/juju-core/+bug/1217282) is from 2013, and the paths that are mentioned don't make too much sense to me (apparently things moved around quite a bit over the years). Note that it's in `won't fix` status, @jameinel might have more info.